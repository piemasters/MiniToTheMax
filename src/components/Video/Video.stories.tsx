import type { Meta, StoryObj } from '@storybook/react-vite';

import { Video } from './Video';

const meta: Meta<typeof Video> = {
  component: Video,
  argTypes: {
    videoId: {
      name: 'videoId',
      description: 'The id part of the source for the video file',
      defaultValue: '',
      control: { type: 'text' },
    },
    title: {
      name: 'title',
      description: 'Title of the video',
      defaultValue: 'Video Title',
      control: { type: 'text' },
    },
    aspectWidth: {
      name: 'aspectWidth',
      description: 'The aspect ratio width of the video',
      defaultValue: 16,
      control: { type: 'number' },
    },
    aspectHeight: {
      name: 'aspectHeight',
      description: 'The aspect ratio height of the video',
      defaultValue: 9,
      control: { type: 'number' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Video>;

export const Primary: Story = {
  args: {
    videoId: 's1Ck5Xn6420',
    title: 'Grey Knight Librarian',
    aspectWidth: 16,
    aspectHeight: 9,
  },
};
