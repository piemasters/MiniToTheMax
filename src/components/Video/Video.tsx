import type { FC } from 'react';
import { css, useTheme } from '@emotion/react';

import { Theme } from '../styles/theme';

export interface VideoProps {
  src: string;
  title: string;
  width?: number;
  aspectRatio?: number;
}

export const Video: FC<VideoProps> = ({
  src,
  title,
  width = 714,
  aspectRatio = 16 / 9,
}) => {
  const theme = useTheme() as Theme;

  const videoStyle = css`
    iframe {
      &:hover {
        box-shadow: 6px 6px 60px 5px ${theme.colors.backgroundTransparent};
      }
    }
  `;
  return (
    <div css={videoStyle} data-testid="video">
      <iframe
        src={src}
        title={title}
        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
        frameBorder="0"
        allowFullScreen
        width={'100%'}
        height={width / aspectRatio}
      />
    </div>
  );
};
export default Video;
