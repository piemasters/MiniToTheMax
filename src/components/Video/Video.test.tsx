import { getByTestId, render } from '@testing-library/react';

import { Video } from './Video';

describe('Video', () => {
  test('renders correctly', () => {
    const { container } = render(
      <Video videoId="qcWaykRRDfM" title="Grey Knight Stormraven" />
    );
    expect(getByTestId(container as HTMLElement, 'video')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
