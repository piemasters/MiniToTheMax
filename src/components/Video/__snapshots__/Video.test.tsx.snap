// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Video renders correctly 1`] = `
<div>
  <div
    class="w-full max-w-3xl hover:shadow-3xl"
    data-testid="video"
  >
    <link
      as="image"
      href="https://i.ytimg.com/vi/qcWaykRRDfM/hqdefault.jpg"
      rel="preload"
    />
    <article
      class="yt-lite "
      data-title="Grey Knight Stormraven"
      style="background-image: url("https://i.ytimg.com/vi/qcWaykRRDfM/hqdefault.jpg"); --aspect-ratio: 56.25%;"
    >
      <button
        aria-label="Watch Grey Knight Stormraven"
        class="lty-playbtn"
        type="button"
      />
    </article>
  </div>
</div>
`;
